src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,193 @@
+
+            String updatedDescription = isCompleted.equals("1") ? "0 " + description : "1 " + description;
+
+            //Reading lines of the file and appending them to StringBuffer
+            while (sc.hasNextLine()) {
+                String nextLine = sc.nextLine();
+                if (nextLine.equals(taskData)) {
+                    buffer.append(updatedDescription + "\n");
+                } else {
+                    buffer.append(nextLine + "\n");
+                }
+            }
+
+            String fileContents = buffer.toString();
+            sc.close();
+            //instantiating the FileWriter class
+            FileWriter writer = new FileWriter("duke.txt");
+            writer.append(fileContents);
+            writer.close();
+        }
+        catch (Exception e) {
+            throw new DukeException();
+        }
+    }
+
+    /**
+     * Deletes task from hard drive.
+     * @param taskData A String that identifies the task in hard drive.
+     * @throws DukeException when there is an error performing the operation.
+     */
+    public void deleteTask(String taskData) throws DukeException {
+        try {
+            //Instantiating the File class
+            File myObj = new File("duke.txt");
+            //Instantiating the Scanner class to read the file
+            Scanner sc = new Scanner(myObj);
+
+            StringBuffer buffer = new StringBuffer();
+
+            //Reading lines of the file and appending them to StringBuffer
+            while (sc.hasNextLine()) {
+                String nextLine = sc.nextLine();
+                if (nextLine.equals(taskData)) {
+
+                } else {
+                    buffer.append(nextLine + "\n");
+                }
+
+            }
+            String fileContents = buffer.toString();
+
+            sc.close();
+
+            //instantiating the FileWriter class
+            FileWriter writer = new FileWriter("duke.txt");
+
+            writer.append(fileContents);
+            writer.close();
+            System.out.println("updated hard drive!");
+        }
+        catch (Exception e) {
+           throw new DukeException();
+        }
+    }
+
+    /**
+     * Adds a task to hard drive.
+     * @param input Description of task to be added to hard drive.
+     * @throws DukeException when there is an error performing the operation.
+     */
+    public void addTask(String input) throws DukeException {
+        try {
+            //Instantiating the File class
+            File myObj = new File("duke.txt");
+            //Instantiating the Scanner class to read the file
+            Scanner sc = new Scanner(myObj);
+
+            StringBuffer buffer = new StringBuffer();
+
+            //Reading lines of the file and appending them to StringBuffer
+            while (sc.hasNextLine()) {
+                buffer.append(sc.nextLine() + "\n");
+            }
+            String fileContents = buffer.toString();
```

Comment:
Seems like theres multiple repeition with calling the files here, you might want to consider storing it under a single method and check for the task type instead?